src/config.rs

```diff
@@ -1,14 +1,74 @@
+use std::{
+    env, fs,
+    io::{self, ErrorKind, Write},
+    path::Path,
+};
 use structopt::StructOpt;
-use telluricdeckay::cli_options;
-use telluricdeckay::config_h;
+use telluricdeckay::{cli_options, config_h};
 
-pub fn get_filename(opt_cfg: Option<String>) -> String {
+fn get_homedir() -> io::Result<String> {
+    let homedir: String = match dirs::home_dir() {
+        Some(homedir) => homedir.to_str().unwrap().into(),
+        None => {
+            return Err(io::Error::new(
+                ErrorKind::NotFound,
+                "Unable to determine homedir",
+            ))
+        }
+    };
+    Ok(homedir)
+}
+
+// Write all values stored in c_data to a config file. This can be used
+// to create the initial config file, and also, later when the UI supports
+// changing options, the existing config file can be overwritten with new
+// values.
+fn write_config(f: &str, c_data: &Data) -> std::io::Result<()> {
+    let mut w = Vec::new();
+    writeln!(&mut w, "PlayerNick = {}", c_data.player_nick)?;
+    writeln!(&mut w, "Server.{}", c_data.is_server)?;
+    writeln!(&mut w, "Server.port = {}", c_data.server_port)?;
+    writeln!(&mut w, "max.players = {}", c_data.max_players)?;
+    writeln!(&mut w, "max.bet = {}", c_data.max_bet)?;
+    writeln!(&mut w, "max.raises = {}", c_data.max_raises)?;
+
+    fs::write(&f, &w)
+}
+
+// Get the path and filename to the config file.
+// If it wasn't given as a cli argument,
+// first check for its existence in the current directory; if not found there,
+// check to see if $XDG_CONFIG_HOME is set, if not, use homedir.
+fn get_filename(opt_cfg: Option<String>, c_data: &Data) -> String {
     if opt_cfg.is_none() {
-        return "./telluricdeckayrc".to_string();
+        let basename = "telluricdeckayrc".to_owned();
+        let cwd = "./".to_owned();
+        let default = cwd + &basename;
+        if std::path::Path::new(&default).exists() {
+            return default;
+        }
+        let config_home = match env::var("XDG_CONFIG_HOME") {
+            Ok(val) => val,
+            Err(_e) => get_homedir().unwrap() + "/.config",
+        };
+        if !Path::new(&config_home).exists() {
+            println!("Creating {}", &config_home);
+            fs::create_dir_all(&config_home).expect("Unable to create config directory");
+        }
+
+        let config_file = config_home + "/" + &basename;
+        if !std::path::Path::new(&config_file).exists() {
+            write_config(&config_file, c_data).expect("Error writing config file");
+        }
+
+        return config_file;
     }
     opt_cfg.unwrap()
 }
 
+// I think the comment below is outdated. Is there any good reason that
```

Comment:
I suspect that comment was for something else. I think it's fine to delete

- [ ] delete lines 69-72.